### DISCUSSION TITLE
VX3 CAN Bus Verbindung zu meinem Raspberry Pi mit Home Assistant OS

### DISCUSSION THREAD

**Discussion by FRMB6** [open3e/open3e#105] [Q&A]
Ich brauch Hilfe mit der Installation von open3e auf meinem Raspberry Pi mit HomeAssistant OS.

Ich habe das Bus Kabel von meinem VX3 zu dem CAN-USB Adapter verbunden, der Adapter hängt wiederum an dem Raspberry Pi.

Wenn ich die weiteren Schritte wie hier im Wiki befolgen möchte funktioniert das aber nicht, weil viele Terminal Befehle im HAOS nicht verfügbar sind. Weiß hier jemand Rat?

Als Lösung würde mir einfallen einen separaten Raspberry Pi nur für das Daten empfangen zu installieren, wobei das imho eigentlich ein Overkill wäre. Ich bin technisch fit, aber kein IT'ler.

Ich würde nach erfolgreicher Integration das auch mit ins Wiki aufnehmen, das ist ja bestimmt für einige interessant.

**Comment by SanMiggel** [comment]
Dazu müsstest du ein eigenes HA add-on erstellen. Das läuft in einem eigenen docker Container und erlaubt die Installation von beliebigen Paketen. Die Frage ist nur ob  der Kernel alle funktionieren unterstützt.
Hatte das nicht schon Mal jemand versucht?

**Reply by FRMB6** [reply]
OK, danke, dann habe ich es ja in etwa richtig verstanden was das Problem ist. Ein eigenes AddOn zu entwickeln liegt denke ich außerhalb meiner Möglichkeiten, wobei das für das Projekt open3e sehr interessant wäre.

**Reply by philippoo66** [reply]
@FRMB6  im Wiki steht was zu [open3e mit Home Assistant](https://github.com/open3e/open3e/wiki/090-Homeassistant) - hast du das schon gesehen?

**Reply by FRMB6** [reply]
Ja, das habe ich schon gesehen, das bezieht sich auf das Datensammeln via MQTT in HomeAssistant. Ich muss ja überhaupt erstmal open3e auf meinem HA Raspberry Pi zum Laufen kriegen.

Gibt es also technisch keinerlei Möglichkeit das auf mein HA Raspberry Pi zu machen? Ist ja irgendwie nicht Sinn der Sache dafür einen anderen Raspberry Pi daneben zu stellen. Und wie gesagt, ich bin kein IT'ler, ich kann leider dafür kein AddOn schreiben.

**Reply by SanMiggel** [reply]
Ich fürchte anders wird es nicht gehen. Das ist genau das Konzept von HAOS, also allgemeine Befehle stark einschränken und zusätzliche Funktionen in Container auslagern. 
Eine Möglichkeit wäre noch auf HAOS zu verzichten und HA auf raspian laufen lassen ist aber auch nicht schön.

Ein add-on zu erstellen ist aber auch kein Hexenwerk, das Projekt bietet eine sehr gute Anleitung: 
https://developers.home-assistant.io/docs/add-ons/tutorial/
Im `Dockerfile` müssten die Befehle zur Installation stehen im `run.sh` der Befehl zum Starten von open3e. Irgendwann wollte ich mir das auch Mal anschauen, aber mein HAOS steht zwei Stockwerke über der Heizung....
